src/analysis/retail/evoker/shared/modules/normalizers/LivingFlamePrePullNormalizer.ts

```diff
@@ -0,0 +1,86 @@
+import { isFromBurnout } from 'analysis/retail/evoker/devastation/modules/normalizers/CastLinkNormalizer';
+import SPELLS from 'common/SPELLS/evoker';
+import { AnyEvent, BeginCastEvent, EventType } from 'parser/core/Events';
+import EventsNormalizer from 'parser/core/EventsNormalizer';
+import Haste from 'parser/shared/modules/Haste';
+import { isRealCast } from 'parser/shared/normalizers/Channeling';
+
+/**
+ * Living flame is a very common pre-pull cast
+ * however it often occurs that you'll start channeling it pre-pull and only
+ * the cast success event will trigger, this messes with how we handle GCDs and makes for a messy timeline
+ * (PrePullNormalizer doesn't account for this since we don't have a way to define instant cast spells)
+ *
+ * This normalizer aims to correct those situation */
+class LivingFlamePrePullNormalizer extends EventsNormalizer {
+  static dependencies = {
+    ...EventsNormalizer.dependencies,
+    haste: Haste,
+  };
+  protected haste!: Haste;
+  normalize(events: AnyEvent[]) {
+    const fixedEvents: AnyEvent[] = events;
+
+    fixedEvents.some((event, eventIdx) => {
```

Comment:
use of `some` here is awkward. i understand why but :thinking:

Reply:
yea looking at it again this should just have been a `for const`, I'll change it 👍